esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -4419,7 +4419,7 @@ void CWsWorkunitsEx::addThorSlaveLogfile(Owned<IConstWorkUnit>& cwu, WsWuInfo& w
         return;
     }
 
-    unsigned numberOfSlaves = clusterInfo->getSize();
+    unsigned numberOfSlaves = clusterInfo->getNumberOfSlaves();
```

Comment:
I think you will need both numberOfSlaves from getSize() and numberOfLogs from getNumberOfLogs() to know which log file to add.